mmv1/third_party/terraform/website/docs/guides/version_5_upgrade.html.markdown

```diff
@@ -96,28 +96,99 @@ Not all of Google cloud resources support labels and annotations. Please check t
+
 The new labels model will be applied to all of the resources with the top level `labels` field or the nested `labels` field inside the top level `metadata` field. Some labels fields are for child resources, so the new model will not be applied to labels fields for child resources.
 
 There are now three label-related fields with the new model:
 
-* The `labels` field will be non-authoritative and only manage the labels defined by the users on the resource through Terraform.
-* The output-only `effective_labels` will list all of labels present on the resource in GCP, including the labels configured through Terraform, other clients and services.
+* The `labels` field will be non-authoritative and only manage the labels defined by the users on the resource through Terraform. If a label was added outside of Terraform, it will not be managed by Terraform, unless it is added to the `labels` field in the configuration. The out of band labels will be listed in the `effective_labels` field. Also, the labels with an empty value (`""`) should be avoided, as they will be ignored and not included in `terraform_labels`, `effective_labels` and then API requests. Instead, labels with the value `_` or `true` are recommended.
 * The output-only `terraform_labels` will merge the labels defined by the users on the resource through Terraform and the default labels configured on the provider. If the same label exists on both the resource labels and provider default labels, the label on the resource will override the provider label.
+* The output-only `effective_labels` will list all of labels present on the resource in GCP, including the labels configured through Terraform, other clients and services.
+
+**Note:** `ignore_changes` can be applied to `labels` field to ignore the changes of the user defined labels. It is not recommended to apply `ignore_changes` to `terraform_labels` or `effective_labels`, as it will make the new labels model not work as expected.
 
-After upgrading to `5.0.0`, and then running `terraform refresh` or `terraform apply`, these three fields should show in the state file of the resources with a self-applying `labels` field.
+The following changes will be observed after upgrading to `5.0.0`.
+* After upgrading to `5.0.0`, and then running `terraform import`, these three fields should show in the state file of the resources with empty `labels` and empty `terraform_labels`. `effective_labels` will have all of labels present on the resource in GCP.
+* After upgrading to `5.0.0`, and then running `terraform refresh`, these three fields should show in the state file of the resources with a self-applying `labels` field and empty `terraform_labels`. `effective_labels` will have all of labels present on the resource in GCP.
```

Comment:
"with a self-applying `labels` field" what does this mean?

I think we expect to see the current set of labels in state preserved as the value for `labels`, right?

Reply:
that is right. Changed.